hibernate-core/src/main/java/org/hibernate/query/sqm/tree/domain/SqmPolymorphicRootDescriptor.java

```diff
@@ -112,7 +91,9 @@ private static boolean isACommonAttribute(List<EntityDomainType<?>> subList, Per
 	}
 
 	public Set<EntityDomainType<?>> getImplementors() {
-		return new HashSet<>( implementors );
+		return implementors.stream()
+				.sorted(Comparator.comparing(EntityDomainType::getTypeName))
+				.collect(Collectors.toCollection(LinkedHashSet::new));
```

Comment:
IMO the sorting should happen during construction of the `SqmPolymoprhicRootDescriptor` (in `org.hibernate.metamodel.model.domain.internal.JpaMetamodelImpl#resolveEntityReference`), since that is cached, by e.g. using a `TreeSet`

```suggestion
		return implementors;
```